src/workerd/api/crypto/scrypt.c++

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "impl.h"
+#include <workerd/api/crypto/kdf.h>
+#include <openssl/crypto.h>
+
+namespace workerd::api {
+
+kj::Maybe<kj::Array<kj::byte>> scrypt(size_t length,
+                                      uint32_t N,
+                                      uint32_t r,
+                                      uint32_t p,
+                                      uint32_t maxmem,
+                                      kj::ArrayPtr<const kj::byte> pass,
+                                      kj::ArrayPtr<const kj::byte> salt) {
+  auto buf = kj::heapArray<kj::byte>(length);
```

Comment:
Node also checks the size of pass and salt here (probably not needed) and applies some error handling by examining the last error if present after calling `EVP_PBE_scrypt()` (which we should adopt too).

Reply:
Length check will happen on the ts side.

Reply:
LGTM, I must have missed the length checks before. Error handling works too.